lib/src/main/java/com/dynatrace/metric/util/Normalize.java

```diff
@@ -54,6 +54,8 @@
   private static final Pattern re_dv_controlCharacters = Pattern.compile("[\\p{C}]+");
   private static final Pattern re_dv_controlCharactersStart = Pattern.compile("^[\\p{C}]+");
   private static final Pattern re_dv_controlCharactersEnd = Pattern.compile("[\\p{C}]+$");
+  private static final Pattern re_dv_hasOddNumberOfTrailingBackslashes =
+      Pattern.compile("[^\\\\](?:\\\\\\\\)*\\\\$");
```

Comment:
Yeah the double-escaping in Java strings doesn't help here. Without that it would be `[^\\](\\\\)*\\$` and your reading is correct.
We could add a clarifying comment but would have to take care so it doesn't go out of sync if this was ever changed :-)
We had the easier regex and explicit counting in 1333e33 before. Both should be fine.

Reply:
That's exactly what its doing. Java doesn't have verbatim strings, that's why one has to escape the regexes so ridiculously. 

I tried that approach first, its rather easy to see in this commit: https://github.com/dynatrace-oss/dynatrace-metric-utils-java/commit/1333e338d1db769effef7275e940e2ff7950019c. I guess both of them work. I would assume the version with the longer regex to be a tiny bit faster, since it only checks the regex and doesn't have to do any other calculations, but in turn the regex is longer, so who knows.